neurom/apps/cut_plane_detection.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2015, Ecole Polytechnique Federale de Lausanne, Blue Brain Project
+# All rights reserved.
+#
+# This file is part of NeuroM <https://github.com/BlueBrain/NeuroM>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import logging
+import math
+import operator
+
+import numpy as np
+import neurom as nm
+
+from neurom import viewer
+from neurom.core import Tree
+from neurom.core.dataformat import COLS
+
+L = logging.getLogger(__name__)
+
+'''Module for the cut plane detection'''
+
+def _create_1d_distributions(points, bin_width):
+    '''Create point distribution integrated along each axis
+
+    Parameters:
+        points: a np.ndarray of points
+        bin_width: the bin width
+
+    Returns: a dict of the X, Y and Z 1D histograms'''
+    min_, max_ = np.min(points, axis=0), np.max(points, axis=0)
+    hist = dict()
+    for i, plane in enumerate('XYZ'):
+        bins = np.arange(math.floor(min_[i]), math.floor(max_[i]) + 1, bin_width)
+        hist[plane] = np.histogram(points[:, i], bins=bins)
+    return hist
+
+
+def _get_probabilities(hist):
+    '''Returns -log(p) where p is the a posteriori probabilities of the observed values
+    in the bins X min, X max, Y min, Y max, Z min, Z max
+
+    Parameters:
+        hist: a dict of the X, Y and Z 1D histograms
+
+    Returns: a dict of -log(p) values'''
+    def compute_p(h):
```

Comment:
more descriptive variable names are preferred over single letter ones except for indices, and the 'special' case of x, y, z axis